py3status/modules/gazelle.py

```diff
@@ -0,0 +1,171 @@
+            raise
+        try:
+            self._cookiejar.save(ignore_discard=True)
+        except IOError:
+            self.py3.log(
+                    "Cannot write cookie at %s" % self.cookiepath,
+                    level=self.py3.LOG_WARNING
+                    )
+        except AttributeError:
+            pass  # That means that no cookie pass has been set
+        else:
+            self.py3.log(
+                    "Created cookie at %s" % self.cookiepath,
+                    level=self.py3.LOG_INFO
+                    )
+
+    def _index(self):
+        resp = self.py3.request(
+                self.baseurl+"/ajax.php",
+                params={'action': "index"},
+                headers={
+                    'User-Agent': 'Mozilla/5.0'
+                    },
+                cookiejar=self._cookiejar
+                )
+        return resp.json()
+
+    def post_config_hook(self):
+        if self.cookiepath:
+            self._cookiejar = MozillaCookieJar(self.cookiepath)
+            try:
+                self._cookiejar.load(ignore_discard=True)
+                self._index()  # just to try if saved cookie is good
+            except IOError:
+                self.py3.log(
+                        "Cannot read cookie at %s" % self.cookiepath,
+                        level=self.py3.LOG_INFO
+                        )
+                self._login()
+            except RequestException:
+                self.py3.log(
+                        "Cookie invalid or outdated. Re-logging.",
+                        level=self.py3.LOG_INFO
+                        )
+                self._login()
+        else:
+            self._cookiejar = CookieJar()
+
+    def gazelle(self):
+        index = self._index()
+        if index['status'] != "success":
+            msg = "Tracker failed to give me user info"
+            self.py3.log(msg, level=self.py3.LOG_ERROR)
+            raise RequestException(msg)
+        result = index['response']
+        data = {
+            'ratio': result['userstats']['ratio'],
+            'requiredratio': result['userstats']['requiredratio'],
+            'download': self._readable_size(result['userstats']['downloaded']),
+            'upload': self._readable_size(result['userstats']['uploaded']),
+            'messages': result['notifications']['messages'],
+            'notifs': result['notifications']['notifications']
+            }
+        if data['ratio'] >= self.limit_warning:
+            color = self.py3.COLOR_RATIO_OK or self.py3.COLOR_GOOD
+        elif data['ratio'] >= (self.limit_required or data['requiredratio']):
+            color = self.py3.COLOR_RATIO_WARNING or self.py3.COLOR_DEGRADED
+        else:
+            color = self.py3.COLOR_RATIO_KO or self.py3.COLOR_BAD
+        full_text = self.py3.safe_format(self.format, data)
+        return {
+            'full_text': full_text,
+            'cached_until': self.py3.time_in(self.cache_timeout),
+            'color': color
```

Comment:
For more flexible customization in coloring, we should remove this `color` line and use `thresholds` instead. The new `format` should be `'\?color=ratio {ratio}'` when it's done correctly.

Also, omit blank line `line 56` for consistency/QA purpose.

Reply:
See my previous comment about thresholds.